src/apiCall.js

```diff
@@ -42,233 +44,356 @@ module.exports = function apiCall(file, apiPort, itApi = it) {
-              body: allReqData,
-              basicAuth,
+          allReqData = evaluateFaker(allReqData, fakerScopes.test);
+          callBefore(apiPort, config, operations, allReqData, basicAuth, req);
+
+          if (utilConditions) {
+            tryer({
+              action(done) {
+                const scResponse = SuperClient(
+                  apiPort,
+                  req,
+                  operations[req.call],
+                  allReqData,
+                  basicAuth,
+                  true,
+                )
+                scResponse.then((response) => {
+                  res = response;
+                  res = addJsonToRes(res);
+
+                  const isJsonExpectMeet = apiPort.expectObject(res.json, untilExpects.json, true)
+                  const headersExpect = _.get(untilExpects, 'headers', false);
+                  let isHeadersExpectMeet = true;
+                  if (headersExpect) {
+                    isHeadersExpectMeet = apiPort.expectObject(res.header, headersExpect, true)
+                  }
+                  if (isJsonExpectMeet && isHeadersExpectMeet) {
+                    isUntilConditionTrue = true;
+                  }
+                  done();
+                }).catch(() => { done() });
+              },
+              pass: () => {
+                try {
+                  successRes(apiPort, config, operations, allReqData, basicAuth, save, req, res, isResPrint);
+                  testDone();
+                } catch (err) {
+                  testDone(err);
+                }
+              },
+              until: () => isUntilConditionTrue,
+              interval,
+              limit,
             });
-          }
-          const res = await SuperClient(apiPort, req, operations[req.call], allReqData, basicAuth)
-
-          if (req.after && _.isFunction(req.after)) {
-            req.after.call(req, {
+          } else {
+            SuperClient(
               apiPort,
-              specs: config.apiCalls.swagger,
-              res,
-              expect,
-              op: operations[req.call],
-              body: allReqData,
-              error: {},
+              req,
+              operations[req.call],
+              allReqData,
               basicAuth,
+            ).then((response) => {
+              res = response;
+              res = addJsonToRes(res);
+              try {
+                successRes(apiPort, config, operations, allReqData, basicAuth, save, req, res, isResPrint);
+                testDone();
+              } catch (err) {
+                testDone(err);
+              }
+            }).catch((err) => {
+              try {
+                callAfter(apiPort, config, operations, allReqData, basicAuth, req, res, true, err);
+                printRes(isResPrint, err, 'Error= ');
+                saveErrorResItem(apiPort, save, isResPrint);
+                assertExpectForError(apiPort, req, err);
+                testDone();
```

Comment:
This block of code repeats here, in the line `189` and  and similarly in the ```successRes``` method. Can we make them all once reused method?

Reply:
Yeah, u r right. We can. Moved to `errorRes()`. :)